include/gpi.h

```diff
@@ -119,7 +119,7 @@ void gpi_get_sim_precision(int32_t *precision);
 // Should be freed with gpi_free_handle
 gpi_sim_hdl gpi_get_root_handle(const char *name);
 gpi_sim_hdl gpi_get_handle_by_name(gpi_sim_hdl parent, const char *name);
-gpi_sim_hdl gpi_get_handle_by_index(gpi_sim_hdl parent, uint32_t index);
+gpi_sim_hdl gpi_get_handle_by_index(gpi_sim_hdl parent, int32_t index);
```

Comment:
I don't know what valid ranges are in Verilog, but this was kind of future looking for VHDL.  When dealing with sfixed, ufixed, and float data types, the range can go negative.

sfixed(3 downto -12) represents a S(4.12) fixed point number.

Further, on custom types that are left unconstrained, the user specifies the range type which could be integer rather than natural or positive, which would then allow for negative indices.